### ISSUE TITLE
Kill Homeserver process when Cypress test is stopped manually

### ISSUE THREAD

**Issue by luixxiul** [vector-im/element-web#25117]
#### What would you like to do?

I would like to see a Homeserver process ended and killed automatically when a Cypress test was manually stopped before `cy.stopHomeserver()` is called on `afterEach`.

#### Why would you like to do it?

Cypress's GUI provides a stop button, and you can stop a test manually by pushing it before reaching to the end of the test. It does not kill the Homeserver process because `cy.stopHomeserver()` does not run yet, so the process stays alive until you kill it by yourself. This sometimes becomes an issue when you stop and restart a test multiple times (while you debug the test, for example), because it creates a new Homeserver process on each time you restart the test, and the memory is eventually filled with the processes and the system becomes unstable.

While the same thing happens by forcing stop a test with <kbd>ctrl</kbd> + <kbd>c</kbd> when you run a test on Terminal, the stop button on GUI version might give an expression that the test was successfully ended and the Homeserver process which has been created for it was also cleared, though it is not. Also, Cypress starts a process for a next test so fast that clicking the stop button just after a test finished seems to be almost impossible to stop a new Homeserver process from being created.

It is possible to kill those processes with `pkill -f` on Terminal manually, but ideally a Homeserver process should be removed when you manually stop the test.

#### How would you like to achieve it?

I am not sure. Would it work to change a setting of Cypress to kill the Homeserver process automatically when you stop the test manually before reaching to its end? I guess checking an existing process and close it if any before starting a Homeserver process with `cy.startHomeserver()` would be fine too.

**Comment by luixxiul** [comment]
The same issue seems to happen when an error occurs, as `afterEach` is not loaded in this case.

**Comment by weeman1337** [comment]
Thank you for your report.

It seems you're not the only one to have come across this bug, as it's been reported previously. I'm going to close this issue now so we can focus on the older issue. Please watch, vote or comment on that issue if there is any additional information you can provide.

**Comment by weeman1337** [comment]
Duplicate of https://github.com/vector-im/element-web/issues/22762
